shotgun_api3/shotgun.py

```diff
@@ -4470,6 +4470,27 @@ def _translate_filters_simple(sg_filter):
     if len(values) == 1 and isinstance(values[0], (list, tuple)):
         values = values[0]
 
+    # Payload optimization: Do not send a full object
+    # just send the `type` and `id` when combining related queries
+    enable_entity_optimization = os.environ.get(
+        "SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION", "0"
+    ).strip().lower()
+    if (
+        enable_entity_optimization not in ["0", "no", "false"]
```

Comment:
Minor point but I'd prefer:
```suggestion
    enable_entity_optimization = os.environ.get(
        "SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION", "0"
    ).strip().lower() in ["1", "yes", "true"]
    if (
        enable_entity_optimization
```

Because:

1. Thinking about it, I prefer that we set to true when the setting value is clearly indicates a "true" value (instead of interpreting "is not false")
2. The first line goal becomes more clear as "read the preference and obtain a boolean value" while the second line is pure algorithm "if true then"
  * In the current code, I feel that the second line (if) does 2 different things: 1. convert the pref to boolean and 2. if true then enable...




Then I'd move the comment above (what this feature does) in the IF block (first lines of the block). But this might only be a personal preference